hilbert/src/main/scala/org/locationtech/sfcurve/hilbert/HilbertCurve2D.scala

```diff
@@ -88,7 +88,7 @@ class HilbertCurve2D(resolution: Int) extends SpaceFillingCurve2D {
     (x, y)
   }
 
-  def toRanges(xmin: Double, ymin: Double, xmax: Double, ymax: Double): Seq[(Long, Long)] = {
+  def toRanges(xmin: Double, ymin: Double, xmax: Double, ymax: Double, maxRecurse: Int = -1 /* not used */): Seq[(Long, Long, Boolean)] = {
```

Comment:
Basically, maxRecursion short circuits the range computation.  This leads to fewer ranges with larger potential over selectivity - but it's a knob that can be adjusted depending on the use case.  I fixed the interface bleeding by using a hinting mechanism.